File: problems/stocks/solutions/bad.py
def get_max_profit(stocks):
    max_profit = stocks[1] - stocks[0]
    current_min = 9999999999999999

    for i, stock in enumerate(stocks):
        if stock < current_min:
            current_min = stock
            for stock_2 in stocks[i + 1:]:
                max_profit = max(max_profit, stock_2 - current_min)

    return max_profit

stocks = [int(x) for x in input().split()]

print(get_max_profit(stocks))

File: problems/stocks/solutions/bad.js
const getMaxProfit = input => {
    const stocks = input.split(' ').map(price => parseInt(price));

    let maxProfit = stocks[1] - stocks[0];
    let currentMin = Number.MAX_VALUE;

    for (let i = 0; i < stocks.length; i++) {
        if (stocks[i] < currentMin) {
            currentMin = stocks[i];
            for (let j = i + 1; j < stocks.length; j++) {
                maxProfit = Math.max(maxProfit, stocks[j] - currentMin);
            }
        }
    }

    return maxProfit;
}



const chunks = [];
process.stdin.on('data', d => chunks.push(d));

process.stdin.on('end', () => {
    const data = chunks.join('').trim();
    console.log(getMaxProfit(data));
});